tensorpack/tfutils/argscope.py

```diff
@@ -97,9 +97,14 @@ def wrapped_func(*args, **kwargs):
         name = func.__name__ if 'name' not in kwargs else kwargs['name']
         if log_shape:
             if ('tower' not in ctx.ns_name.lower()) or ctx.is_main_training_tower:
+                # we assume the first parameter is the most intersting
```

Comment:
If it's really going to handle arbitrary user function, it's best to (1) document what types of functions is supported in the documentation and (2) add a check for it (either skip the logging or raise an exception, both seems OK)